numpyro/distributions/continuous.py

```diff
@@ -1188,6 +1188,70 @@ def infer_shapes(
         return batch_shape, event_shape
 
 
+class CAR(Distribution):
+    arg_constraints = {
+        "loc": constraints.real_vector,
+        "alpha": constraints.unit_interval,
+        "tau": constraints.positive,
+        "W": constraints.positive_definite,
+    }
+    support = constraints.real_vector
+    reparametrized_params = [
+        "loc",
+        "alpha",
+        "tau",
+    ]
+
+    def __init__(
+        self,
+        loc=0.0,
+        alpha=None,
+        tau=None,
+        W=None,
+        validate_args=None,
+    ):
+        if jnp.ndim(loc) == 0:
+            (loc,) = promote_shapes(loc, shape=(1,))
+        # temporary append a new axis to loc
+        loc = loc[..., jnp.newaxis]
+        if W is not None:
+            loc, self.W = promote_shapes(loc, W)
+
+        batch_shape = lax.broadcast_shapes(jnp.shape(loc)[:-2], jnp.shape(self.W)[:-2])
```

Comment:
I think you can use
```
# no need: loc = loc[..., jnp.newaxis]
batch_shape = lax.broadcast_shapes(jnp.shape(loc)[:-1], jnp.shape(alpha), jnp.shape(tau), jnp.shape(self.W)[:-2])
event_shape = jnp.shape(self.W)[-1:]
self.loc = promote_shapes(loc, shape=batch_shape + event_shape)
self.alpha, self.tau = promote_shapes(alpha, tau, shape=batch_shape)
```

Reply:
the shape stuff is very confusing but I'll rely on your experience. Let me know if I implemented your suggestion correctly when I push these commits